src/Control/Distributed/Spark/SQL/StructField.hs

```diff
@@ -1,7 +1,9 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE UndecidableInstances #-}
```

Comment:
What's `UndecidableInstances` for?

Reply:
It's because GHC doesn't like the `SingI (Ty a)` constraint in `Coercible`. It says the context is no smaller than the instance head so is worried about non-termination. This is also true for `Reify` and `Reflect` instances in general. Maybe future versions of GHC will have a more discriminating termination checker.